pages/plugins.rst

```diff
@@ -81,160 +108,70 @@ If you want to continue working on the command line, you can do the following to
-
-Throw a ``ConfigurationException`` if the user should have entered missing or invalid configuration parameters.
-
-Registering the plugin
-----------------------
-
-.. _registering_alarm_callback:
-
-You now have to register your plugin in the ``JiraAlarmCallbackModule.java`` file to make ``graylog-server`` load the alarm callback when launching. The
-reason for the manual registering is that a plugin could consist of multiple plugin types. Think of the generated plugin file as a bundle of
-multiple plugins.
-
-Register your new plugin using the ``configure()`` method::
-
-  @Override
-  protected void configure() {
-      addAlarmCallback(JiraAlarmCallback.class);
-  }
-
-.. _writing_decorators:
-
-Writing a decorator plugin
-==========================
-
-Writing a custom decorator is generally similar to the aforementioned alarm callback. After :ref:`creating_plugin_skeleton`, you need to implement a class implementing the ``SearchResponseDecorator`` interface. This class must:
+.. note:: A single plugin can contain multiple extensions to Graylog.
+  
+  For example a hypothetical plugin might contribute an input, an output and alert notifications to communicate with systems. For convenience this would be bundled in a single plugin registering multiple extensions.
 
-  * Contain interfaces extending the ``SearchResponseDecorator.Factory``, ``SearchResponseDecorator.Config`` & ``SearchResponseDecorator.Descriptor`` interface
-  * Implement a ``apply`` method, containing the actual logic of decorating a message
+Required classes
+................
 
-The ``Factory`` interface needs a bit more explanation. It works as a connection between the factory, config, description and actual implementation classes. It should look similar to this::
+At the very minimum you need to implement two interfaces:
 
-      public interface Factory extends SearchResponseDecorator.Factory {
-        @Override
-        YourMessageDecorator create(Decorator decorator);
+  * ``org.graylog2.plugin.Plugin`` - which is the entry to your `plugin <https://github.com/Graylog2/graylog-plugin-sample/blob/2.2/src/main/java/org/graylog/plugins/sample/SamplePlugin.java>`_ code
+  * ``org.graylog2.plugin.PluginMetaData`` - which `describes your plugin <https://github.com/Graylog2/graylog-plugin-sample/blob/2.2/src/main/java/org/graylog/plugins/sample/SampleMetaData.java>`_
 
-        @Override
-        YourMessageDecorator.Config getConfig();
+The ``bootstrap-plugin`` script generates these implementations for you, and you simply need to fill out the details.
 
-        @Override
-        YourMessageDecorator.Descriptor getDescriptor();
-      }
+Graylog uses Java's ServiceLoader mechanism to find your plugin's main class, so if you rename your ``Plugin`` implementation, you need to also adjust the `service file <https://github.com/Graylog2/graylog-plugin-sample/blob/2.2/src/main/resources/META-INF/services/org.graylog2.plugin.Plugin>`_.
```

Comment:
Maybe link to the ServiceLoader documentation: https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html
Maybe link to Google Auto Service (used by some of our own plugins): https://github.com/google/auto/tree/master/service